docs/source/user-guide/nlp/nlp_metadata.rst

```diff
@@ -0,0 +1,80 @@
+Metadata can be anything - from the source of the data, the timestamp of its creation, to the age of the author, and
+any other information that is given about the text.
+
+What Is Metadata Used For?
+=============================
+
+Metadata is used by Deepchecks' checks to extract meaningful features from the data, since some computations cannot be
+computed directly on the text (for example, drift). Inspecting the distribution of a metadata column's values (e.g. to
+notice that the source of the texts is different between the train and test sets) can help uncover potential problems in
+the way that the datasets were built, or hint about the model's expected performance on unseen data.
+
+Example for specific scenarios in which measuring the metadata may come in handy:
+
+#. **Find weak segments** - The metadata can be used to segment the data and test for low performing segments.
+   If found, the weak segment may indicate an underrepresented segment or an area where the data quality is worse.
+#. **Find obscure relations between the data and the targets** - the model training might be affected
+   by the metadata, even if it doesn't have access to it.
+   For example, in a classification dataset of toxic statements, if the toxic statements came from a specific source
+   (e.g. Twitter), the model might learn to predict the label by other traits of the source (e.g. the allowed length of
+   the text). In this case, the model will perform well on the training data, and may even perform well on the test
+   data, but will fail to generalize to new data.
+
+
+How To Use Metadata
+=====================
+
+Metadata Format
+---------------
+Metadata must be given as a pandas DataFrame, with the rows representing each sample and columns representing the
+different metadata columns. The number of rows in the metadata DataFrame must be equal to the number of samples in the
+dataset, and the order of the rows must be the same as the order of the samples in the dataset.
+
+How To Pass Metadata To Deepchecks
+-----------------------------------
+In order to pass metadata to Deepchecks, you must pass it to the :class:`TextData <deepchecks.nlp.TextData>` object.
+The metadata can be passed to the :class:`TextData <deepchecks.nlp.TextData>` object in two ways:
+
+#. When initializing the :class:`TextData <deepchecks.nlp.TextData>` object, by passing the metadata to the
+   ``metadata`` parameter.
+#. After initializing the :class:`TextData <deepchecks.nlp.TextData>` object, by using the ``TextData.set_metadata``
+   function.
+
+In both options, it's advised to also use the ``categorical_metadata`` parameter to specify which metadata columns are
+categorical. The parameter should be a list of the names of the categorical columns.
+
+
+In the following example, we will pass the metadata to the ``TextData`` object in order to use the
+``MetadataSegmentsPerformance`` check:
+
+.. code-block:: python
```

Comment:
Consider showing a printout of a metadata dataframe

Reply:
This is an rst file.

Reply:
Can add a table to rst

Reply:
Seems weird and unnecessary IMO... Is it important to you?

Reply:
don't invest in it more than 5 minutes, but I bet you gpt4 can do that in 1.

Reply:
My problem is not that it would take time, it's that I don't think it would look good.